Storage/test.cc

```diff
@@ -1002,23 +1021,70 @@ TEST(TransactionalStorage, RESTfulAPITest) {
     EXPECT_EQ(200, static_cast<int>(HTTP(DELETE(base_url + "/api/data/post/test")).code));
     EXPECT_EQ(404, static_cast<int>(HTTP(GET(base_url + "/api/data/post/test")).code));
 
+    // Test RESTful matrix too.
+    EXPECT_EQ(404, static_cast<int>(HTTP(GET(base_url + "/api/data/like/dima-beer")).code));
+    EXPECT_EQ(404, static_cast<int>(HTTP(GET(base_url + "/api/data/like/max-beer")).code));
+
+    // For this test, we disallow POST for the `/like`-s matrix.
+    EXPECT_EQ(405, static_cast<int>(HTTP(POST(base_url + "/api/data/like", SimpleLike("dima", "beer"))).code));
+
+    // Add a few likes.
+    EXPECT_EQ(
+        201,
+        static_cast<int>(HTTP(PUT(base_url + "/api/data/like/dima-beer", SimpleLike("dima", "beer"))).code));
+    EXPECT_EQ(201,
+              static_cast<int>(
+                  HTTP(PUT(base_url + "/api/data/like/max-beer", SimpleLike("max", "beer", "Cheers!"))).code));
+
+    EXPECT_EQ(200, static_cast<int>(HTTP(GET(base_url + "/api/data/like/dima-beer")).code));
+    EXPECT_EQ(200, static_cast<int>(HTTP(GET(base_url + "/api/data/like/max-beer")).code));
+
+    EXPECT_EQ("{\"row\":\"dima\",\"col\":\"beer\",\"details\":null}\n",
+              HTTP(GET(base_url + "/api/data/like/dima-beer")).body);
+    // Meh, this test is not for `AdvancedHypermedia`. -- D.K.
+    // EXPECT_EQ("{\"row\":\"max\",\"col\":\"beer\"}\n",
+    //           HTTP(GET(base_url + "/api/data/like/max-beer?fields=brief")).body);
+    EXPECT_EQ("{\"row\":\"max\",\"col\":\"beer\",\"details\":\"Cheers!\"}\n",
+              HTTP(GET(base_url + "/api/data/like/max-beer")).body);
+
+    // GET matrix as the collection.
+    EXPECT_EQ(200, static_cast<int>(HTTP(GET(base_url + "/api/data/like")).code));
+    EXPECT_EQ("max:beer\ndima:beer\n", HTTP(GET(base_url + "/api/data/like")).body);
```

Comment:
Yeah, legacy of our separation between `Rest/{Basic,Hypermedia/AdvancedHypermedia}`.

In a nutshell, JSON is not a strict requirement for REST, and the unit test for `Basic`, non-JSON, REST, is cleaner to read.

We'll probably revisit this some time soon.